File: src/main/java/org/vstu/compprehension/models/entities/EnumData/Role.java
package org.vstu.compprehension.models.entities.EnumData;

import com.fasterxml.jackson.annotation.JsonProperty;

public enum Role{
    @JsonProperty("STUDENT")
    STUDENT,
    @JsonProperty("TEACHER")
    TEACHER,
    @JsonProperty("ADMIN")
    ADMIN;
}

File: modules/server/src/main/js/types/role.ts
import * as io from 'io-ts'

export type Role = "STUDENT" | "TEACHER" | "ADMIN"
export const TRole: io.Type<Role> = io.keyof({
    "STUDENT": null,
    "TEACHER": null,
    "ADMIN": null,
}, 'Role')
